Assets/LeapMotionModules/UI/Interaction/InteractionBehaviourBase.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Leap;
+using Leap.Unity.Attributes;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  public abstract class InteractionBehaviourBase : MonoBehaviour {
+
+    public InteractionManager interactionManager;
+
+    protected Rigidbody _body;
+    public Rigidbody Rigidbody { get { return _body; } }
+
+    [SerializeField]
+    [HideInInspector]
+    #pragma warning disable 0414
+    // Variable used only for inspector rendering.
+    private bool _interactionManagerIsNull = true;
+    #pragma warning restore 0414
+
+    [Header("Interaction Settings")]
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreHover = false; // TODO: 
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreContact = false; // TODO: Contact NYI.
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreGrasping = false;
+    /// <summary> Can this object be grasped with two or more hands? </summary>
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool allowMultiGrasp = false;
+
+    /// <summary>
+    /// Called by the InteractionManager every FixedUpdate, after
+    /// InteractionHands have FixedUpdated.
+    /// 
+    /// InteractionBehaviour uses this to fire per-object interaction events.
+    /// </summary>
+    public abstract void FixedUpdateObject();
+
+    protected virtual void Awake() {
+      _body = GetComponent<Rigidbody>();
+      _body.maxAngularVelocity = 100F;
+
+      if (interactionManager != null) {
+        interactionManager.RegisterInteractionBehaviour(this);
+      }
+    }
+
+    protected virtual void Start() {
+      if (interactionManager == null) {
+        interactionManager = InteractionManager.singleton;
+
+        if (interactionManager == null) {
+          Debug.LogError("Interaction Behaviours require an Interaction Manager. Please ensure you have an InteractionManager in your scene.");
+        }
+        else {
+          interactionManager.RegisterInteractionBehaviour(this);
+        }
+      }
+    }
+
+    protected virtual void OnValidate() {
+      _interactionManagerIsNull = interactionManager == null;
+      _body = GetComponent<Rigidbody>();
+    }
+
+    /// <summary> Return the distance the interaction object is from the given world position. </summary>
+    public abstract float GetDistance(Vector3 worldPosition);
+
+    #region Hovering
+
+    /// <summary> Called per-hand when that hand produces a non-zero hover score for this object. </summary>
+    public abstract void HoverBegin(Hand hand);
```

Comment:
Does this mean that objects that are currently being grasped can get hover callbacks?  Can they get hover callbacks from the hand that is currently grasping them?  If so, is there a guarantee that the hand currently grasping the object has it's primary hover set to be this object as well?